# Data Quality Assessment

## Objective

Evaluate the quality of the Myntra dataset by checking for:

- Missing values
- Duplicate records
- Invalid values
- Incorrect data types
- Outliers
- Data consistency

The goal is to ensure the dataset is reliable before performing analysis.

## Import Required Libraries

Import the libraries required for data quality assessment.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Load Dataset

Load the Myntra dataset into a Pandas DataFrame.

In [2]:
df = pd.read_csv("../data/myntra_dataset_ByScraping.csv")

## Verify Dataset

Display the first five rows to ensure that the dataset has been loaded correctly.

In [3]:
df.head()

,brand_name,pants_description,price,MRP,discount_percent,ratings,number_of_ratings
0,WROGN,Men Loose Fit Cotton Jeans,1374.0,2499.0,0.45,4.2,57.0
1,Flying Machine,Men Slim Fit Jeans,1829.0,2999.0,0.39,4.6,5.0
2,Roadster,Men Pure Cotton Jeans,974.0,2499.0,0.61,3.6,1100.0
3,Bene Kleed,Relaxed Fit Denim Jeans,873.0,2299.0,0.62,4.0,4800.0
4,Levis,Men 511 Slim Fit Jeans,1478.0,2899.0,0.49,4.3,264.0


## Data Quality Check 1 — Missing Values

### Business Question

Are there any missing values in the dataset?

Missing values can affect calculations such as averages, correlations, and visualizations. Identifying them is the first step in any data quality assessment.

In [4]:
df.isnull().sum()

brand_name           0
pants_description    0
price                0
MRP                  0
discount_percent     0
ratings              0
number_of_ratings    0
dtype: int64

### Observation

No missing values were found in any column of the dataset.

This indicates that all 52,120 product records contain complete information for brand name, product description, price, MRP, discount percentage, ratings, and number of ratings.

No data imputation or removal of missing records is required.

## Data Quality Check 2 — Missing Value Percentage

Knowing the percentage of missing values helps determine whether data should be removed or imputed.

In [5]:
(df.isnull().sum()/len(df))*100

brand_name           0.0
pants_description    0.0
price                0.0
MRP                  0.0
discount_percent     0.0
ratings              0.0
number_of_ratings    0.0
dtype: float64

### Observation

The percentage of missing values is 0% across all columns.

Since every feature contains complete data, the dataset is suitable for further analysis without requiring missing value treatment.

## Data Quality Check 3 — Duplicate Rows

### Business Question

Are there duplicate records?

Duplicate records can inflate product counts and produce misleading business insights.

In [6]:
df.duplicated().sum()

np.int64(17047)

### Observation

A total of 17,047 duplicate rows were identified in the dataset.

This is a significant number of duplicates and requires further investigation before removal.

Since this is an e-commerce dataset, duplicate rows may represent products listed multiple times due to different sizes, colors, or repeated scraping. Therefore, these records should not be removed without verifying whether they are true duplicates.

## Data Quality Check 4 — Duplicate Product Descriptions

Some product descriptions may repeat because products are available in different sizes or colors.

We will check how many descriptions are repeated before deciding whether any action is needed.

In [7]:
df["pants_description"].duplicated().sum()

np.int64(49440)

### Observation

A total of **49,440 duplicate product descriptions** were found.

This result is expected in an e-commerce dataset because the same product description can be associated with multiple product variants such as different sizes, colors, or repeated listings.

Since the dataset does not contain attributes like size or color, duplicate product descriptions should **not** be considered data quality issues.

Therefore, no records will be removed based solely on duplicate product descriptions.

## Data Quality Check 5 — Verify Data Types

Correct data types are important because numerical analysis can only be performed on numeric columns.

In [8]:
df.dtypes

brand_name               str
pants_description        str
price                float64
MRP                  float64
discount_percent     float64
ratings              float64
number_of_ratings    float64
dtype: object

### Observation

The dataset contains both categorical and numerical variables.

Categorical columns:
- brand_name
- pants_description

Numerical columns:
- price
- MRP
- discount_percent
- ratings
- number_of_ratings

The data types are appropriate for analysis and do not require conversion.

## Data Quality Check 6 — Validate Price

Business Rule:

Product price cannot be zero or negative.

In [9]:
(df["price"] <= 0).sum()

np.int64(0)

### Observation

Invalid prices found: 0

## Data Quality Check 7 — Validate MRP

Business Rule:

MRP should always be greater than zero.

In [10]:
(df["MRP"] <= 0).sum()

np.int64(0)

### Observation

No products were found with an MRP less than or equal to zero.

All products have valid original prices.

## Data Quality Check 8 — Price vs MRP

Business Rule

Selling Price should never exceed the MRP.

In [11]:
(df["price"] > df["MRP"]).sum()

np.int64(0)

### Observation

No products were found where the selling price exceeds the MRP.

This confirms that the pricing information follows the expected business rule and appears to be internally consistent.

No corrective action is required.

## Data Quality Check 9 — Ratings

Business Rule

Ratings must be between 0 and 5.

In [12]:
((df["ratings"] < 0) | (df["ratings"] > 5)).sum()

np.int64(0)

### Observation

All product ratings fall within the valid range of 0 to 5.

No invalid customer rating values were detected.

## Data Quality Check 10 — Discount Validation

Business Rule

Discount percentage should be between 0 and 1 because the dataset stores discount as a decimal value.

In [13]:
((df["discount_percent"] < 0) | (df["discount_percent"] > 1)).sum()

np.int64(4117)

### Observation

A total of **4,117 records** have discount values outside the expected decimal range of 0–1 (i.e., values greater than 1).

These values (ranging from 1.1 to 64.0) reflect an **inconsistent scraper format** — not standard percentage decimals (0.45) or whole percentages that can be reliably converted (e.g., dividing by 100 does not match price-derived discounts).

This issue will be addressed during the data cleaning phase.

### Observation

No products were found with a negative number of customer ratings.

All values in the `number_of_ratings` column satisfy the expected business rule and are valid for further analysis.

No corrective action is required.